oak-run/src/main/js/oak-mongo.js

```diff
@@ -329,6 +329,59 @@ var oak = (function(global){
         return {nRemoved : count};
     };
 
+    /**
+     * Helper method to find nodes based on Regular Expression.
+     *
+     * @memberof oak
+     * @method regexFind
+     * @param {string} pattern the pattern to match the nodes.
+     */
+	api.regexFind = function(pattern) {
+		print(db.nodes.find({_id: {$regex: pattern}}));
+		db.nodes.find({_id: {$regex: pattern}}).forEach(function(doc) {
+			print(doc._id);
+        });
+	}
+
+    /**
+     * Remove the complete subtree of all the nodes matching a regex pattern.
+     * Use regexFind to find the nodes that match the pattern prior deletion.
+     *
+     * @memberof oak
+     * @method removeDescendantsMatching
+     * @param {string} pattern the pattern to match the nodes to be removed.
+     */
+    api.removeDescendantsMatching = function(pattern) {
```

Comment:
```suggestion
    api.removeDescendantsAndSelfMatching = function(pattern) {
```

naming : should we use a similar name to `removeDescendantsAndSelf` to stress that `self` will also be deleted?

Reply:
Done